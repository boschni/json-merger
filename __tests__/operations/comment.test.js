const jsonMerger = require("../../dist");
const { testConfig } = require("../../__helpers__");

describe("when merging two objects and a source property has a $comment indicator", function () {
  test("it should not merge the $comment property", function () {
    const object1 = {
      a: {
        aa: "original",
      },
    };

    const object2 = {
      a: {
        $comment: "comment",
      },
    };

    const result = jsonMerger.mergeObjects([object1, object2], testConfig());

    expect(result).toMatchSnapshot();
  });
});
