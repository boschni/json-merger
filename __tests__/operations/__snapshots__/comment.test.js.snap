// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when merging two objects and a source property has a $comment indicator it should not merge the $comment property 1`] = `
"{
	\\"a\\": {
		\\"aa\\": \\"original\\"
	}
}"
`;
